src/main/java/org/alliancegenome/curation_api/services/helpers/GenericOntologyLoadHelper.java

```diff
@@ -91,13 +91,15 @@ public T traverse(OWLClass parent, int depth) throws Exception {
 
             termParent = getOntologyTerm(parent);
 
-            String requiredNamespace = config.getAltNameSpace();
-            if (requiredNamespace == null) {
-                requiredNamespace = defaultNamespace;
+            ArrayList<String> requiredNamespaces = config.getAltNameSpaces();
+            if (requiredNamespaces.isEmpty()) {
+                if (defaultNamespace != null) {
+                    requiredNamespaces.add(defaultNamespace);
+                }
```

Comment:
This block should happen once outside the traverse method, as it will get called on every node. Should go around line 69 or so.

This is an interesting idea with the list of alt name spaces. I just wonder if it could get confusing, where like in the ZFA file there are two ontologies (ZFA, ZFS), and two namespaces, vs a single ontology with multiple namespaces (GO). I believe this will work, I guess lets see what happens.

Reply:
done